frontend/scripts/runE2eWithServer.ts

```diff
@@ -0,0 +1,105 @@
+import tcpPortUsed from 'tcp-port-used'
+import path from 'path'
+import process from 'process'
+
+const PORT = {
+  FE: 4321,
+  BE: 4322,
+  STUB: 4323,
+}
+
+const DIR = {
+  FE: path.resolve(__dirname, '../'),
+  BE: path.resolve(__dirname, '../../backend'),
+  STUB: path.resolve(__dirname, '../../stubs'),
+}
+
+const main = async (args: string[]) => {
+  const { $ } = await import('execa')
+
+  const checkPorts = async () => {
+    const ports = Object.values(PORT)
+    const checkPorts = ports.map((port) => tcpPortUsed.check(port))
+
+    const usingPorts = await Promise.all(checkPorts).then((result) => ports.filter((_, idx) => result[idx]))
+    const availablePorts = ports.filter((port) => !usingPorts.includes(port))
+
+    console.log(`E2E: Currently available ports: ${availablePorts.join(',')}`)
```

Comment:
- the prefix `E2E:` is not necessary
- follow our log convention, update to `Successfully to do ....` at least